hack/test-go.sh

```diff
@@ -148,38 +155,33 @@ if [[ -n "${KUBE_COVER}" ]]; then
   # each of the coverage profiles generated when running 'go test -cover', but
   # we strip these lines out when combining so that there's only one.
   echo "mode: ${KUBE_COVERMODE}" >${combined_cover_profile}
-fi
 
-if [[ -n "${1-}" ]]; then
-  test_dirs=$@
-else
-  test_dirs=$(kube::test::find_dirs)
-fi
-
-# Run all specified tests, optionally collecting coverage if KUBE_COVER is set.
-for arg in ${test_dirs}; do
-  trap 'exit 1' SIGINT
-  pkg=${KUBE_GO_PACKAGE}/${arg}
+  # Run all specified tests, optionally collecting coverage if KUBE_COVER is set.
+  for arg in ${test_dirs}; do
+    pkg=${KUBE_GO_PACKAGE}/${arg}
 
-  cover_params=()
-  cover_profile=""
-  if [[ -n "${KUBE_COVER}" ]]; then
     cover_profile=${cover_report_dir}/${arg}/coverage.out
     mkdir -p "${cover_report_dir}/${arg}"
     cover_params=(-cover -covermode="${KUBE_COVERMODE}" -coverprofile="${cover_profile}")
-  fi
 
+    go test "${goflags[@]:+${goflags[@]}}" \
+        ${KUBE_RACE} \
+        ${KUBE_TIMEOUT} \
+        "${cover_params[@]+${cover_params[@]}}" \
+        "${pkg}"
+    if [[ -f "${cover_profile}" ]]; then
+      # Include all coverage reach data in the combined profile, but exclude the
+      # 'mode' lines, as there should be only one.
+      grep -h -v "^mode:" ${cover_profile} >>${combined_cover_profile} || true
+    fi
+  done
+else
+  dirs=(${test_dirs})
   go test "${goflags[@]:+${goflags[@]}}" \
       ${KUBE_RACE} \
       ${KUBE_TIMEOUT} \
-      "${cover_params[@]+${cover_params[@]}}" \
-      "${pkg}"
-  if [[ -f "${cover_profile}" ]]; then
-    # Include all coverage reach data in the combined profile, but exclude the
-    # 'mode' lines, as there should be only one.
-    grep -h -v "^mode:" ${cover_profile} >>${combined_cover_profile} || true
-  fi
-done
+      $(printf "${KUBE_GO_PACKAGE}/%s " "${dirs[@]:+${dirs[@]}}")
```

Comment:
Use a substitution expression instead on the array. If I'm correct, the actual list will be in `$@` so you'd have:

```shell
  go test \
      ... \
      ${KUBE_TIMEOUT} \
      "${@+${@/#/${KUBE_GO_PACKAGE}/}}"
```

If it's in the `testcases` array instead:

```shell
  go test \
      ... \
      ${KUBE_TIMEOUT} \
      "${testcases[@]+${testcases[@]/#/${KUBE_GO_PACKAGE}/}}"
```